### PR TITLE
feat(public-cloud): add Local Zone

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | #MANAGER-12742 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

As a customer,

- I would like to create instances in new local zones
- I would like to have a listing view of all my existing instances
- I would like to edit my localzone instance
- I would like to create a new "localzone" private network
- I would like to list private networks in two distinct tabs (typical private network / local private mode)

## Related

- [x] TRANS CMT-7236

### CODE REVIEW

File: packages/manager/modules/pci/src/components/project/flavors-list/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -19,6 +20,9 @@
 
   "pci_project_flavors_price_hourly": "{{price}} HT/heure",
   "pci_project_flavors_price_monthly": "À partir de {{price}} HT/mois",
+  "pci_project_flavors_zone_compatible": "Compatible with",
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/components/project/flavors-list/translations/Messages_fr_FR.json:23]:
```suggestion
  "pci_project_flavors_zone_compatible": "Compatible avec",
```

File: packages/manager/modules/pci/src/components/project/regions-list/regions.service.js [hunk 2]
```diff
@@ -1,65 +1,52 @@
...
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+          Accept: 'application/json',
+          'X-Pagination-Mode': 'CachedObjectList-Pages',
+          'X-Pagination-Size': 5000,
+        },
+      },
+    );
+
+    return response.json();
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/components/project/regions-list/regions.service.js:27]: We can use this.iceberg

File: packages/manager/modules/pci/src/components/project/regions-list/regions.service.js [hunk 3]
```diff
@@ -1,65 +1,52 @@
...
+    return this.$q
+      .all({
+        regions: this.getAvailableRegions(serviceName),
+        quotas: fetch(`/engine/api/cloud/project/${serviceName}/quota`, {
+          method: 'GET',
+          credentials: 'same-origin',
+          headers: {
+            'Content-Type': 'application/json;charset=utf-8',
+            Accept: 'application/json',
+          },
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/components/project/regions-list/regions.service.js:40]: We can use $http.get

File: packages/manager/modules/manager-components/src/region/translations/Messages_en_GB.json
Comment (ghyenne) [inline packages/manager/modules/manager-components/src/region/translations/Messages_en_GB.json:1]: It's en_GB, maybe should be fr_FR to modify

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 4]
```diff
@@ -54,11 +54,30 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
                     data-heading="{{:: continent }}"
                 >
                     <div class="container-fluid px-0">
+                        <p>Global regions ... Oles ipsum...</p>
```
[the remark this answers is not in the document]
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html:57]: Its takend cared in next PR.

File: packages/manager/modules/manager-components/src/region/translations/Messages_en_GB.json
Comment (sachinrameshn) [reply packages/manager/modules/manager-components/src/region/translations/Messages_en_GB.json:1]: I have removed it in next PR.

File: packages/manager/modules/pci/src/components/project/regions-list/regions.service.js [hunk 5]
```diff
@@ -1,65 +1,39 @@
...
-          ),
-        ).then((regionsDetail) => compact(regionsDetail)),
-      );
+  async getAvailableRegions(serviceName) {
+    return this.iceberg(`/cloud/project/${serviceName}/region`)
+      .query()
+      .expand('CachedObjectList-Pages')
+      .execute()
+      .$promise.then(({ data }) => data)
+      .catch(() => []);
```
Comment (darsene) [inline packages/manager/modules/pci/src/components/project/regions-list/regions.service.js:20]: no need to display any error ?

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 6]
```diff
@@ -54,11 +54,30 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
...
                         <div class="row">
                             <oui-select-picker
                                 class="d-inline-block col-md-6 col-xl-3 my-3"
                                 data-ng-repeat="(location, datacenters) in datacentersLocation track by location"
-                                data-ng-if="datacenters.length"
+                                data-ng-if="datacenters.length && location !== 'Madrid'"
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:62]: `Madrid` check is only for mocking data ? If yes, add a TODO to not forget to remove it

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 7]
```diff
@@ -305,6 +324,7 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
         </div>
 
         <div
+            data-ng-show="false"
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:327]: What's the utility ?
```suggestion
```

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html [hunk 8]
```diff
@@ -381,14 +401,15 @@ <h1 data-translate="pci_projects_project_instances_add_title"></h1>
...
+                data-disabled="mode.disabled"
             >
                 <oui-select-picker-section>
                     <p data-translate="{{:: mode.description1}}"></p>
                     <p data-translate="{{:: mode.description2}}"></p>
                 </oui-select-picker-section>
             </oui-select-picker>
         </div>
-        <div>
+        <div data-ng-show="false">
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.html:412]: What's the utility ?
```suggestion
        <div>
```

File: packages/manager/modules/pci/src/projects/project/instances/add/translations/Messages_fr_FR.json [hunk 9]
```diff
@@ -44,6 +44,9 @@
   "pci_projects_project_instances_add_privateNetwork_private_mode": "Mode Privé",
   "pci_projects_project_instances_add_privateNetwork_private_mode_description1": "Les instances seront uniquement attachées à un réseau privé. Il peut être soit complètement privé, soit exposé à Internet en utilisant les services Gateway ou Load Balancer avec les Floating IP.",
   "pci_projects_project_instances_add_privateNetwork_private_mode_description2": "Veuillez noter que pour un accès SSH depuis internet, une Floating IP ou une autre instance (appelée SSH-Proxy) est nécessaire.",
+  "pci_projects_project_instances_add_privateNetwork_local_private_mode": "Local Private mode",
+  "pci_projects_project_instances_add_privateNetwork_local_private_mode_description1": "description1",
+  "pci_projects_project_instances_add_privateNetwork_local_private_mode_description2": "description2",
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/instances/add/translations/Messages_fr_FR.json:49]: Do not forget to update translation values

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html (hunk 7 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html:327]: Its been removed in next PR.  https://github.com/ovh/manager/pull/10523

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html (hunk 8 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html:412]: Its already been take cared in next PR: https://github.com/ovh/manager/pull/10523

File: packages/manager/modules/pci/src/projects/project/instances/add/add.html (hunk 6 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.html:62]: I have removed it here : https://github.com/ovh/manager/pull/10523

**Review by darsene — APPROVED** [review]

File: packages/manager/modules/pci/src/components/project/regions-list/regions.service.js (hunk 5 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/components/project/regions-list/regions.service.js:20]: No need here.

**Review by qpavy — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**